text/0032-distinct-attribute.md

```diff
@@ -0,0 +1,238 @@
+### III. Additional Materials
+
+#### Algolia
+
+Algolia distinct feature is based on one attribute, as defined in `attributeForDistinct`. It is possible to limit the number of returned records that contains the same attribute value. It is done at indexing time.
+
+Algolia distinct functionality enables deduplication and aggregation by allowing a numeric value for the defined distinct attribute.
+
+| Behavior       | Distinct value | Description                                                                                       |
+|----------------|----------------|---------------------------------------------------------------------------------------------------|
+| de-duplication | N = 1          | Used to remove similar records from the search result. Only the most relevant record is returned. |
+| grouping       | N > 1          | N records containing the same value for the distinct attribute will be returned.                  |
+
+> `distinct` is silently ignored at query time if `attibuteForDistinct` is not defined. It is not mandatory but possible to set `distinct` at indexing time via the index settings endpoint.
+>
+> It is possible to disable distinct feature at query time, by giving a falsy value (`false` or `O`) for the distinct parameter. Giving `true` is equivalent to `1`.
+
+#### TypeSense
+
+TypeSense distinct feature uses `group_by` and `group_limit` to achieve de-duplication and grouping at query time.
+
+> A field that is setted as a parameter of group_by must be previously faceted.
+
+| Parameter   | Description                                                                                                                                                                                                                                   |
+|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| group_by    | It is possible to aggregate records into groups by setting multiple fields separated by a comma. E.g. group_by=country,company_name                                                                                                           |
+| group_limit | Control the maximum number of top records returned for groups. By default, TypeSense `group_limit` parameter is set to 3.                                                                                                                     |
+
+Using group_by add a nested structure in the search result. Buckets are returned in `grouped_hits` field.
+
+#### ElasticSearch
+
+Elasticsearch does not provide this functionality directly. Nor a keyword or an operator exists to get de-duplicated or grouped results.
+
+It is difficult to find the information in the official documentation. A lot of people are asking about the distinct feature on StackOverflow or ElasticSearch forums.
```

Comment:
Their documentation seems to be voluntarily hard

Reply:
This can be explained by the fact that they do not target a particular application with Elasticsearch. The documentation is much more precise for their Enterprise products which are scoped on specific use cases.